src/lib/SequentialBucket.ts

```diff
@@ -0,0 +1,52 @@
+type RatelimitedFunction = () => Promise<void>;
+
+const wait = (ms: number) => new Promise((rs) => setTimeout(rs, ms));
+
+// TODO(TTtie): test this in the real world
+class SequentialBucket {
+  private queue: RatelimitedFunction[] = [];
+  public remaining: number = 1;
+  public limit: number = 1;
+  private timeout: number = 0;
+  public lastTime: number = 0;
+  public lastLocalTime: number = 0;
+  public resetOn: number = 0;
+
+  private processing: boolean = false;
```

Comment:
You do not need to specify these types, they are redundant as they are inferred by the assignment. Please try not to do this in future.

Reply:
that makes sense but I prefer explicit typing

Reply:
It's kinda wasteful don't you think?

Reply:
it's my preferrence though

Reply:
But why? Surely you would want code to be informative through minimal amounts of information?

Reply:
It's not like these are separate typings for a JS file, we have typescript inferring stuff and other humans who read the code also inferring stuff.

Reply:
thats why i prefer js ngl

Reply:
That's why I prefer not explicitly typing them? It's not necessary. TS complements JS, it only makes it needlessly verbose if you make it do such a thing.

Reply:
bruh its explicitly typed in the regular bucket as well

Reply:
You want me to comment there too?

Reply:
yeah no